src/main/java/commons/headers/Headers.java

```diff
@@ -0,0 +1,85 @@
+package commons.headers;
+
+import commons.header.Header;
+
+import java.util.*;
+
+public class Headers implements HeadersInterface {
+    private final HashMap<String, ArrayList<Header>> hashMap = new LinkedHashMap<>();
+
+    @Override
+    public HeadersInterface add(String key, byte[] value) throws IllegalStateException {
+        Header newHeader = new Header(key, value);
+        // First time keys
+        if (!hashMap.containsKey(key)) {
+            hashMap.put(key, new ArrayList<>());
+            hashMap.get(key).add(newHeader);
+        } else {
+            hashMap.get(key).add(newHeader);
+        }
+        return this;
+    }
+
+    @Override
+    public HeadersInterface add(Header header) throws IllegalStateException {
+        String key = header.getKey();
+        // First time keys
+        if (!hashMap.containsKey(key)) {
+            hashMap.put(key, new ArrayList<>());
+            hashMap.get(key).add(header);
+        } else {
+            hashMap.get(key).add(header);
+        }
+        return this;
+    }
+
+    @Override
+    public HeadersInterface remove(String key) throws IllegalStateException {
+        // Key does not exist
+        if (!hashMap.containsKey(key)) {
+            throw new IllegalArgumentException("Key not found.");
+        }
+
+        hashMap.get(key).clear();
+        return this;
+    }
+
+    @Override
+    public Iterable<Header> headers(String key) {
+        // Key does not exist
+        if (!hashMap.containsKey(key)) {
+            throw new IllegalArgumentException("Key not found.");
+        }
+        return hashMap.get(key);
+    }
+
+    @Override
+    public Header lastHeader(String key) {
+        // Key does not exist
+        if (!hashMap.containsKey(key)) {
+            throw new IllegalArgumentException("Key not found.");
```

Comment:
similar question as https://github.com/tferdous17/flux/pull/29/files#r1898777566

Reply:
I think we should, @tferdous17 what's your input?